_search-plugins/search-pipelines/truncate-hits-processor.md

```diff
@@ -0,0 +1,515 @@
+        "_score" : 1.0,
+        "_source" : {
+          "doc" : {
+            "title" : "document 3"
+          }
+        }
+      },
+      {
+        "_index" : "my_index",
+        "_id" : "4",
+        "_score" : 1.0,
+        "_source" : {
+          "doc" : {
+            "title" : "document 4"
+          }
+        }
+      },
+      {
+        "_index" : "my_index",
+        "_id" : "5",
+        "_score" : 1.0,
+        "_source" : {
+          "doc" : {
+            "title" : "document 5"
+          }
+        }
+      }
+    ]
+  }
+}
+```
+</details>
+
+## Oversample, collapse, and truncate hits
+
+This shows a more realistic example where you would use `oversample` to request many candidate documents, use `collapse` to remove documents that
+duplicate some field (to get more diverse results), then use `truncate` to return the originally-requested document count (avoiding returning a
+large result payload from the cluster).
+
+
+### Setup
+
+Create many documents with a field that we'll use for collapsing:
+
+```json
+POST /_bulk
+{ "create":{"_index":"my_index","_id":1}}
+{ "title" : "document 1", "color":"blue" }
+{ "create":{"_index":"my_index","_id":2}}
+{ "title" : "document 2", "color":"blue" }
+{ "create":{"_index":"my_index","_id":3}}
+{ "title" : "document 3", "color":"red" }
+{ "create":{"_index":"my_index","_id":4}}
+{ "title" : "document 4", "color":"red" }
+{ "create":{"_index":"my_index","_id":5}}
+{ "title" : "document 5", "color":"yellow" }
+{ "create":{"_index":"my_index","_id":6}}
+{ "title" : "document 6", "color":"yellow" }
+{ "create":{"_index":"my_index","_id":7}}
+{ "title" : "document 7", "color":"orange" }
+{ "create":{"_index":"my_index","_id":8}}
+{ "title" : "document 8", "color":"orange" }
+{ "create":{"_index":"my_index","_id":9}}
+{ "title" : "document 9", "color":"green" }
+{ "create":{"_index":"my_index","_id":10}}
+{ "title" : "document 10", "color":"green" }
+``` 
+{% include copy-curl.html %}
+
+Create a pipeline that just collapses on the `color` field:
+
+```json
+PUT /_search/pipeline/collapse_pipeline
+{
+  "response_processors": [
+    {
+      "collapse" : {
+        "field": "color"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Create another pipeline that oversamples, collapses, then truncates results:
+
+```json
+PUT /_search/pipeline/oversampling_collapse_pipeline
+{
+  "request_processors": [
+    {
+      "oversample": {
+        "sample_factor": 3
+      }
+    }
+  ],
+  "response_processors": [
+    {
+      "collapse" : {
+        "field": "color"
+      }
+    },
+    {
+      "truncate_hits": {
+        "description": "Truncates back to the original size before oversample increased it."
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Collapse without oversample
+
+In this example, we request the top 3 documents before collapsing on the "color" field. Since the first 2 documents have the same "color", the second one is discarded,
```

Comment:
```suggestion
In this example, you request the top 3 documents before collapsing on the "color" field. Because the first two documents have the same `color`, the second one is discarded,
```